1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,5 +1,61 @@
-function setAlarm() {}
+function setAlarm() {
+  const input = document.getElementById("alarmSet").value;
+  const setBtn = document.getElementById("set");
+  const stopBtn = document.getElementById("stop");
+  const timer = document.getElementById("timeRemaining");
 
+  // Make Set Alarm Button working:
+  setBtn.addEventListener("click", () => {
+    // check if the input is a positive integer.
+    input <= 0 || input === ""
```

Comment:
Is the value of `input` set every time the button is clicked?
What data type (number, string, array, null, object, Error) is the value of the element going to be?

Reply:
I've set the input type to "number", so it only accepts numbers.

Reply:
The `value` attribute is always text/string as far as I can tell, even if the `type` attribute is set to `number`. So you always need to pass it through `parseInt(input, 10)` and check for `NaN`